sql/catalyst/src/main/scala/org/apache/spark/sql/connector/catalog/CatalogV2Implicits.scala

```diff
@@ -115,7 +109,11 @@ private[sql] object CatalogV2Implicits {
           s"$quoted is not a valid TableIdentifier as it has more than 2 name parts.")
     }
 
-    def quoted: String = parts.map(quote).mkString(".")
+    def quoted: String = quoteNameParts(parts.toArray)
```

Comment:
I don't think this is a good refactor. It introduces a method call and an array allocation just to replace a single line with a single line.